src/utilitis/logger.ts

```diff
@@ -5,24 +5,29 @@ export default class Logger {
     this._verbose = verbose;
   }
 
-  warn(message: string): void {
-    if (this.verbose) console.warn(message);
+  warn(message: string, component: string = 'Generic Component'): void {
+    if (this.verbose) {
+      console.warn(this.formatLogMsg('WARN', component, message));
+    }
   }
 
-  info(message: string): void {
+  info(message: string, component: string = 'Generic Component'): void {
     if (this.verbose) {
-      console.log(message);
+      console.log(this.formatLogMsg('INFO', component, message));
     }
   }
 
-  debug(message: string): void {
+  debug(message: string, component: string = 'Generic Component'): void {
```

Comment:
Nice, it could be nice to have a group or a method that returns a logger that closes on the ```component```.